core/domain/suggestion_services.py

```diff
@@ -0,0 +1,382 @@
+            suggestion_type)]
+
+
+def get_suggestions_by_target_id(target_type, target_id):
+    """Gets a list of suggestions to the entity with the given ID.
+
+    Args:
+        target_type: str. The type of target entity the suggestion is linked to.
+        target_id: str. The ID of the target entity the suggestion is linked to.
+
+    Returns:
+        list(Suggestion). A list of suggestions linked to the entity.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_target_id(
+            target_type, target_id)]
+
+
+def _update_suggestion(suggestion):
+    """Updates the given suggestion.
+
+    Args:
+        suggestion: Suggestion. The domain object of the suggestion to be
+            updated.
+    """
+
+    suggestion_model = suggestion_models.SuggestionModel.get_by_id(
+        suggestion.suggestion_id)
+
+    suggestion_model.status = suggestion.status
+    suggestion_model.assigned_reviewer_id = suggestion.assigned_reviewer_id
+    suggestion_model.final_reviewer_id = suggestion.final_reviewer_id
+    suggestion_model.change_cmd = suggestion.change_cmd
+    suggestion_model.score_category = suggestion.score_category
+
+    suggestion_model.put()
+
+
+def mark_review_completed(suggestion, status, reviewer_id):
+    """Marks that a review has been completed.
+
+    Args:
+        suggestion: Suggestion. The domain object of the suggestion to be
+            updated.
+        status: str. The status of the suggestion post review.
+        reviewer_id: str. The ID of the user who completed the review.
+    """
+    suggestion.status = status
+    suggestion.final_reviewer_id = reviewer_id
+    suggestion.assigned_reviewer_id = None
+
+    _update_suggestion(suggestion)
+
+
+def assign_reviewer_to_suggestion(suggestion, assigned_reviewer_id):
+    """Assigns a user to review the suggestion.
+
+    Args:
+        suggestion: Suggestion. The domain object of the suggestion to be
+            updated.
+        assigned_reviewer_id: str. The ID of the user who is assigned to review.
+    """
+    suggestion.status = suggestion_models.STATUS_IN_REVIEW
+    suggestion.assigned_reviewer_id = assigned_reviewer_id
+
+    _update_suggestion(suggestion)
+
+
+# TODO (nithesh): This function is temporary. At the moment, the feedback
+# threads id is of the form exp_id.<random_str> while the suggestion ids are of
+# the form entity_type.entity_id.<random_str>. Once the feedback thread ID
+# migration is complete, these two IDs will be matched, and this function will
+# be removed.
+def get_thread_id_from_suggestion_id(suggestion_id):
+    """Gets the thread_id from the suggestion_id.
+
+    Args:
+        suggestion_id: str. The ID of the suggestion.
+
+    Returns:
+        str. The thread ID linked to the suggestion.
+    """
+    return suggestion_id[suggestion_id.find('.') + 1:]
+
+
+def is_suggestion_handled(suggestion):
```

Comment:
Seems more like suggestion.is_handled? (define as a `@property`)

Especially since it is primarily dependent on a property of the suggestion object.

Reply:
Done